apis/python/src/tiledbsoma/options/_soma_tiledb_context.py

```diff
@@ -66,25 +70,30 @@ class SOMATileDBContext(ContextBase):
 
     def __init__(
         self,
-        tiledb_config: Optional[Dict[str, Union[str, float]]] = None,
-        timestamp: Optional[OpenTimestamp] = None,
-        threadpool: Optional[ThreadPoolExecutor] = None,
+        tiledb_config: Dict[str, Union[str, float]] | None = None,
+        tiledb_ctx: "tiledb.Ctx" | None = None,
```

Comment:
```suggestion
        tiledb_config: Dict[str, Union[str, float]] | "tiledb.Ctx" | None = None,
```

I would slightly prefer this. Admittedly passing a context to the `tiledb_config` doesn't sound very good, but it eliminates the error case of passing both a context and a config. You could take a leap and say that a context is part of the "configuration".